src/main/java/org/opensearch/security/http/OnBehalfOfAuthenticator.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.http;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.JwtParser;
+import io.jsonwebtoken.security.WeakKeyException;
+import org.apache.hc.core5.http.HttpHeaders;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import org.opensearch.OpenSearchException;
+import org.opensearch.OpenSearchSecurityException;
+import org.opensearch.SpecialPermission;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.rest.RestChannel;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.security.auth.HTTPAuthenticator;
+import org.opensearch.security.authtoken.jwt.EncryptionDecryptionUtil;
+import org.opensearch.security.ssl.util.ExceptionUtils;
+import org.opensearch.security.user.AuthCredentials;
+import org.opensearch.security.util.KeyUtils;
+
+import static org.opensearch.security.OpenSearchSecurityPlugin.LEGACY_OPENDISTRO_PREFIX;
+import static org.opensearch.security.OpenSearchSecurityPlugin.PLUGINS_PREFIX;
+
+public class OnBehalfOfAuthenticator implements HTTPAuthenticator {
+
+    private static final String REGEX_PATH_PREFIX = "/(" + LEGACY_OPENDISTRO_PREFIX + "|" + PLUGINS_PREFIX + ")/" + "(.*)";
+    private static final Pattern PATTERN_PATH_PREFIX = Pattern.compile(REGEX_PATH_PREFIX);
+    private static final String ON_BEHALF_OF_SUFFIX = "api/generateobotoken";
+    private static final String ACCOUNT_SUFFIX = "api/account";
+
+    protected final Logger log = LogManager.getLogger(this.getClass());
+
+    private static final Pattern BEARER = Pattern.compile("^\\s*Bearer\\s.*", Pattern.CASE_INSENSITIVE);
+    private static final String BEARER_PREFIX = "bearer ";
+
+    private final JwtParser jwtParser;
+    private final String encryptionKey;
+    private final Boolean oboEnabled;
+    private final String clusterName;
+
+    public OnBehalfOfAuthenticator(Settings settings, String clusterName) {
+        String oboEnabledSetting = settings.get("enabled");
```

Comment:
You may be able to set a default in this call to avoid the null check on the line below. 

`settings.get("enabled", <default_here>);`

Reply:
If we are considering removing the null check, the implicit assumption we are making is that the "enabled" key will always be present in the settings. If it's not always guaranteed to be present, and we remove the null check, I think it will encounter a NullPointerException when trying to convert a null String to a Boolean. Plz correct me if I'm wrong. I think our original design is to have this feature enable by default even if the user don't configure the 'enabled' field.

Reply:
Good catch, fixed.